# HG changeset patch
# User Rik <rik@octave.org>
# Date 1372625448 25200
#      Sun Jun 30 13:50:48 2013 -0700
# Node ID b04ae15530fc55696879479e0e155711fa41e026
# Parent  f160f9029b4e0396a583c31ae7ab410077064929
Rename default_save_options() to save_default_options().

* NEWS: Announce renaming.

* doc/interpreter/basics.txi, doc/interpreter/io.txi: Update references
in manual to point to save_default_options.

* libgui/src/resource-manager.cc: Update keyword list.

libinterp/interpfcn/load-save.cc(Fsave_default_options): Rename function.  Rename
seealso links.  Rename variable Vdefault_save_options to Vsave_defaut_options.

* libinterp/octave.cc(maximum_braindamage): Call save_default_options.

* libinterp/parse-tree/pt-assign.cc: Update list of old Octave built-in variables.

* scripts/deprecated/default_save_options.m: Add deprecated function which calls
save_default_options.

* scripts/deprecated/module.mk: Add deprecated script default_save_options.m to build.

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -173,23 +173,23 @@ Summary of important user-visible change
 
       helpdlg    listdlg   questdlg
       inputdlg   msgbox    warndlg
 
  ** Other new functions added in 3.8.0:
 
       base64_decode               ellipke         polyeig              
       base64_encode               erfcinv         rgbplot                   
-      betaincinv                  erfi            shrinkfaces            
-      built_in_docstrings_file    expint          splinefit              
-      cmpermute                   findfigs        strjoin                
-      cmunique                    fminsearch      struct2hdl             
-      colorcube                   gallery         tetramesh              
-      copyobj                     gco             waterfall              
-      dawson                      hdl2struct                  
+      betaincinv                  erfi            save_default_options
+      built_in_docstrings_file    expint          shrinkfaces           
+      cmpermute                   findfigs        splinefit             
+      cmunique                    fminsearch      strjoin               
+      colorcube                   gallery         struct2hdl            
+      copyobj                     gco             tetramesh             
+      dawson                      hdl2struct      waterfall             
       dblist                      importdata  
       doc_cache_create            iscolormap
       ellipj                      lines   
                            
 
  ** Deprecated functions.
 
     The following functions were deprecated in Octave 3.4 and have been
@@ -200,16 +200,17 @@ Summary of important user-visible change
       betai      gammai                perror       values
       cellidx    glpkmex               replot               
       cquad      is_duplicate_entry    saveimage            
       
     The following functions have been deprecated in Octave 3.8 and will
     be removed from Octave 3.12 (or whatever version is the second major
     release after 3.8):
 
+      default_save_options
       gen_doc_cache
       java_convert_matrix
       java_debug
       java_get
       java_invoke
       java_new
       java_set
       java_unsigned_conversion
diff --git a/doc/interpreter/basics.txi b/doc/interpreter/basics.txi
--- a/doc/interpreter/basics.txi
+++ b/doc/interpreter/basics.txi
@@ -243,17 +243,17 @@ user preferences to the following values
 @example
 @group
 PS1                             = ">> "
 PS2                             = ""
 allow_noninteger_range_as_index = true
 beep_on_error                   = true
 confirm_recursive_rmdir         = false
 crash_dumps_octave_core         = false
-default_save_options            = "-mat-binary"
+save_default_options            = "-mat-binary"
 do_braindead_shortcircuit_evaluation = true
 fixed_point_format              = true
 history_timestamp_format_string = "%%-- %D %I:%M %p --%%"
 page_screen_output              = false
 print_empty_dimensions          = false
 @end group
 @end example
 
diff --git a/doc/interpreter/io.txi b/doc/interpreter/io.txi
--- a/doc/interpreter/io.txi
+++ b/doc/interpreter/io.txi
@@ -144,17 +144,17 @@ carriage return.
 @node Simple File I/O
 @subsection Simple File I/O
 
 @cindex saving data
 @cindex loading data
 The @code{save} and @code{load} commands allow data to be written to and
 read from disk files in various formats.  The default format of files
 written by the @code{save} command can be controlled using the functions
-@code{default_save_options} and @code{save_precision}.
+@code{save_default_options} and @code{save_precision}.
 
 As an example the following code creates a 3-by-3 matrix and saves it
 to the file @samp{myfile.mat}.
 
 @example
 @group
 A = [ 1:3; 4:6; 7:9 ];
 save myfile.mat A
@@ -175,17 +175,17 @@ A
      @print{}    7   8   9
 @end group
 @end example
 
 @DOCSTRING(save)
 
 There are three functions that modify the behavior of @code{save}.
 
-@DOCSTRING(default_save_options)
+@DOCSTRING(save_default_options)
 
 @DOCSTRING(save_precision)
 
 @DOCSTRING(save_header_format_string)
 
 @DOCSTRING(load)
 
 @DOCSTRING(fileread)
diff --git a/libgui/src/resource-manager.cc b/libgui/src/resource-manager.cc
--- a/libgui/src/resource-manager.cc
+++ b/libgui/src/resource-manager.cc
@@ -658,17 +658,16 @@ resource_manager::octave_keywords (void)
       "debug "
       "debug_on_error "
       "debug_on_interrupt "
       "debug_on_warning "
       "dec2base "
       "dec2bin "
       "dec2hex "
       "deconv "
-      "default_save_options "
       "del2 "
       "delaunay "
       "delaunay3 "
       "delaunayn "
       "delete "
       "dellistener "
       "demo "
       "det "
@@ -1443,16 +1442,17 @@ resource_manager::octave_keywords (void)
       "run "
       "run_count "
       "run_history "
       "run_test "
       "rundemos "
       "runlength "
       "runtests "
       "save "
+      "save_default_options "
       "save_header_format_string "
       "save_precision "
       "saveas "
       "saveaudio "
       "saveimage "
       "saveobj "
       "savepath "
       "saving_history "
diff --git a/libinterp/interpfcn/load-save.cc b/libinterp/interpfcn/load-save.cc
--- a/libinterp/interpfcn/load-save.cc
+++ b/libinterp/interpfcn/load-save.cc
@@ -98,17 +98,17 @@ static bool Vcrash_dumps_octave_core = t
 // write to the Octave core file.
 static double Voctave_core_file_limit = -1.0;
 
 // The name of the Octave core file.
 static std::string Voctave_core_file_name = "octave-workspace";
 
 // The default output format.  May be one of "binary", "text",
 // "mat-binary", or "hdf5".
-static std::string Vdefault_save_options = "-text";
+static std::string Vsave_default_options = "-text";
 
 // The output format for Octave core files.
 static std::string Voctave_core_file_options = "-binary";
 
 static std::string
 default_save_header_format (void)
 {
   return
@@ -1465,17 +1465,17 @@ Save the named variables @var{v1}, @var{
 output to the terminal.  If no variable names are listed, Octave saves\n\
 all the variables in the current scope.  Otherwise, full variable names or\n\
 pattern syntax can be used to specify the variables to save.\n\
 If the @option{-struct} modifier is used, fields @var{f1} @var{f2} @dots{}\n\
 of the scalar structure @var{STRUCT} are saved as if they were variables\n\
 with corresponding names.\n\
 Valid options for the @code{save} command are listed in the following table.\n\
 Options that modify the output format override the format specified by\n\
-@code{default_save_options}.\n\
+@code{save_default_options}.\n\
 \n\
 If save is invoked using the functional form\n\
 \n\
 @example\n\
 save (\"-option1\", @dots{}, \"file\", \"v1\", @dots{})\n\
 @end example\n\
 \n\
 @noindent\n\
@@ -1571,17 +1571,17 @@ The command\n\
 \n\
 @example\n\
 save -binary data a b*\n\
 @end example\n\
 \n\
 @noindent\n\
 saves the variable @samp{a} and all variables beginning with @samp{b} to\n\
 the file @file{data} in Octave's binary format.\n\
-@seealso{load, default_save_options, save_header_format_string, dlmread, csvread, fread}\n\
+@seealso{load, save_default_options, save_header_format_string, dlmread, csvread, fread}\n\
 @end deftypefn")
 {
   octave_value_list retval;
 
   int argc = args.length ();
 
   string_vector argv = args.make_argv ();
 
@@ -1595,17 +1595,17 @@ the file @file{data} in Octave's binary 
 
   load_save_format format = LS_ASCII;
 
   bool append = false;
 
   bool use_zlib = false;
 
   // get default options
-  parse_save_options (Vdefault_save_options, format, append, save_as_floats,
+  parse_save_options (Vsave_default_options, format, append, save_as_floats,
                       use_zlib);
 
   // override from command line
   argv = parse_save_options (argv, format, append, save_as_floats,
                              use_zlib);
   argc = argv.length ();
   int i = 0;
 
@@ -1768,33 +1768,33 @@ When called from inside a function with 
 changed locally for the function and any subroutines it calls.  The original\n\
 variable value is restored when exiting the function.\n\
 @seealso{octave_core_file_limit, octave_core_file_name, octave_core_file_options}\n\
 @end deftypefn")
 {
   return SET_INTERNAL_VARIABLE (crash_dumps_octave_core);
 }
 
-DEFUN (default_save_options, args, nargout,
+DEFUN (save_default_options, args, nargout,
   "-*- texinfo -*-\n\
-@deftypefn  {Built-in Function} {@var{val} =} default_save_options ()\n\
-@deftypefnx {Built-in Function} {@var{old_val} =} default_save_options (@var{new_val})\n\
-@deftypefnx {Built-in Function} {} default_save_options (@var{new_val}, \"local\")\n\
+@deftypefn  {Built-in Function} {@var{val} =} save_default_options ()\n\
+@deftypefnx {Built-in Function} {@var{old_val} =} save_default_options (@var{new_val})\n\
+@deftypefnx {Built-in Function} {} save_default_options (@var{new_val}, \"local\")\n\
 Query or set the internal variable that specifies the default options\n\
 for the @code{save} command, and defines the default format.\n\
 Typical values include @code{\"-ascii\"}, @code{\"-text -zip\"}.\n\
 The default value is @option{-text}.\n\
 \n\
 When called from inside a function with the \"local\" option, the variable is\n\
 changed locally for the function and any subroutines it calls.  The original\n\
 variable value is restored when exiting the function.\n\
 @seealso{save}\n\
 @end deftypefn")
 {
-  return SET_NONEMPTY_INTERNAL_STRING_VARIABLE (default_save_options);
+  return SET_NONEMPTY_INTERNAL_STRING_VARIABLE (save_default_options);
 }
 
 DEFUN (octave_core_file_limit, args, nargout,
   "-*- texinfo -*-\n\
 @deftypefn  {Built-in Function} {@var{val} =} octave_core_file_limit ()\n\
 @deftypefnx {Built-in Function} {@var{old_val} =} octave_core_file_limit (@var{new_val})\n\
 @deftypefnx {Built-in Function} {} octave_core_file_limit (@var{new_val}, \"local\")\n\
 Query or set the internal variable that specifies the maximum amount\n\
diff --git a/libinterp/octave.cc b/libinterp/octave.cc
--- a/libinterp/octave.cc
+++ b/libinterp/octave.cc
@@ -621,17 +621,17 @@ maximum_braindamage (void)
 
   FPS1 (octave_value (">> "));
   FPS2 (octave_value (""));
   FPS4 (octave_value (""));
   Fallow_noninteger_range_as_index (octave_value (true));
   Fbeep_on_error (octave_value (true));
   Fconfirm_recursive_rmdir (octave_value (false));
   Fcrash_dumps_octave_core (octave_value (false));
-  Fdefault_save_options (octave_value ("-mat-binary"));
+  Fsave_default_options (octave_value ("-mat-binary"));
   Fdo_braindead_shortcircuit_evaluation (octave_value (true));
   Ffixed_point_format (octave_value (true));
   Fhistory_timestamp_format_string (octave_value ("%%-- %D %I:%M %p --%%"));
   Fpage_screen_output (octave_value (false));
   Fprint_empty_dimensions (octave_value (false));
 
   disable_warning ("Octave:abbreviated-property-match");
   disable_warning ("Octave:fopen-file-in-path");
diff --git a/libinterp/parse-tree/pt-assign.cc b/libinterp/parse-tree/pt-assign.cc
--- a/libinterp/parse-tree/pt-assign.cc
+++ b/libinterp/parse-tree/pt-assign.cc
@@ -69,17 +69,16 @@ static const char *former_built_in_varia
   "beep_on_error",
   "completion_append_char",
   "crash_dumps_octave_core",
   "current_script_file_name",
   "debug_on_error",
   "debug_on_interrupt",
   "debug_on_warning",
   "debug_symtab_lookups",
-  "default_save_options",
   "echo_executing_commands",
   "fixed_point_format",
   "gnuplot_binary",
   "gnuplot_command_axes",
   "gnuplot_command_end",
   "gnuplot_command_plot",
   "gnuplot_command_replot",
   "gnuplot_command_splot",
@@ -96,16 +95,17 @@ static const char *former_built_in_varia
   "octave_core_file_name",
   "output_max_field_width",
   "output_precision",
   "page_output_immediately",
   "page_screen_output",
   "print_answer_id_name",
   "print_empty_dimensions",
   "print_rhs_assign_val",
+  "save_default_options",
   "save_header_format_string",
   "save_precision",
   "saving_history",
   "sighup_dumps_octave_core",
   "sigterm_dumps_octave_core",
   "silent_functions",
   "split_long_rows",
   "string_fill_char",
diff --git a/scripts/deprecated/default_save_options.m b/scripts/deprecated/default_save_options.m
new file mode 100644
--- /dev/null
+++ b/scripts/deprecated/default_save_options.m
@@ -0,0 +1,42 @@
+## Copyright (C) 2013 Rik Wehbring
+##
+## This file is part of Octave.
+##
+## Octave is free software; you can redistribute it and/or modify it
+## under the terms of the GNU General Public License as published by
+## the Free Software Foundation; either version 3 of the License, or (at
+## your option) any later version.
+##
+## Octave is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Octave; see the file COPYING.  If not, see
+## <http://www.gnu.org/licenses/>.
+
+## -*- texinfo -*-
+## @deftypefn  {Built-in Function} {@var{val} =} default_save_options ()
+## @deftypefnx {Built-in Function} {@var{old_val} =} default_save_options (@var{new_val})
+## @deftypefnx {Built-in Function} {} default_save_options (@var{new_val}, \"local\")
+## This function has been deprecated.  Use @code{@file{save_default_options}}
+## instead.
+## @seealso{save_default_options}
+## @end deftypefn
+
+## Deprecated in 3.8
+
+function retval = default_save_options (varargin)
+
+  persistent warned = false;
+  if (! warned)
+    warned = true;
+    warning ("Octave:deprecated-function",
+             "default_save_options is obsolete and will be removed from a future version of Octave, please use save_default_options instead");
+  endif
+
+  retval = save_default_options (varargin{:});
+
+endfunction
+
diff --git a/scripts/deprecated/module.mk b/scripts/deprecated/module.mk
--- a/scripts/deprecated/module.mk
+++ b/scripts/deprecated/module.mk
@@ -1,15 +1,16 @@
 FCN_FILE_DIRS += deprecated
 
 deprecated_FCN_FILES = \
   deprecated/__error_text__.m \
   deprecated/cor.m \
   deprecated/corrcoef.m \
   deprecated/cut.m \
+  deprecated/default_save_options.m \
   deprecated/java_debug.m \
   deprecated/error_text.m \
   deprecated/gen_doc_cache.m \
   deprecated/isstr.m \
   deprecated/java_convert_matrix.m \
   deprecated/java_get.m \
   deprecated/java_invoke.m \
   deprecated/java_new.m \
